WordPress/gutenberg
# Spinner: Convert component to TypeScript

## What?

Converts the `Spinner` component to TypeScript

Part of #35744

## Why?

Brings us closer to completely typing the components package!

## How?

I converted the relevant files to TS. I changed the story so there is now a default one, and one to allow editing of the height and width in a `Custom Size` story.

## Testing Instructions




1. Run `npm run dev` and ensure no typing errors.
2. Run `npm run storybook:dev` and visit the `Spinner` component. 
3. Ensure you can change the height/width props in the `Custom Size` story and that the Spinner displays correctly at different sizes without the stroke width changing.

## Screenshots or screencast 

cc @ciampo
